### ISSUE TITLE
CustomCertService checkTrusted: use CompleteableDeferred instead of POJO lock/notify

### ISSUE THREAD

**Issue by rfc2822** [bitfireAT/cert4android#20]
To make the library more stable and future-proof, I'd like to get rid of some legacies like AIDL (#12) or in this case POJO lock/notify: 

https://github.com/bitfireAT/cert4android/blob/f167e847cdfd8045c7b15cfaf548cec27a281e24/lib/src/main/java/at/bitfire/cert4android/CustomCertManager.kt#L166-L190

I think a more modern approach would be:

- [ ] `ICustomCertService` should return a `CompletableDeferred<Boolean>` instead of taking the `IOnCertificateDecision` callback
- [ ] In `CustomCertManager`, we could wait for the result like here: https://github.com/bitfireAT/davx5/blob/1430010196a6d0ac4cb109d0d14a329570b3de96/app/src/main/java/at/bitfire/davdroid/syncadapter/SyncAdapterServices.kt#L96-L104
- [ ] `CustomCertService.binder.checkTrusted` should create and use the `CompletableDeferred<Boolean>`

I hope that this will work as I imagine. Can you please give it a try @ArnyminerZ?

**Comment by ArnyminerZ** [comment]
Why don't simply using suspending and Boolean return? Most of the times you will want the current thread to block until a decision is taking, and otherwise you can simply launch a new coroutine and call it a day

**Comment by rfc2822** [comment]
Coroutines and "simply" are a bad match in my opinion ;) But in this case, yes if it works and generates less code, give it a try :)

**Comment by rfc2822** [comment]
Wouldn't this than rather be an application of flows? Well I'll wait to see what you had in mind

**Comment by ArnyminerZ** [comment]
I have finally changed my mind, and switched to cancellable suspending functions. Take a look and tell me if you like that approach, or if I should change to something different. For me it's working fine, and I've added timeout for custom certs.

Note that I've kept the old method for legacy, but it's all deprecated. Maybe we want to remove all the methods now, or wait to some point in the future. I don't know how many people is using the library
